spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -318,16 +353,31 @@ public final void addQueues(Queue... queues) {
 	public boolean removeQueueNames(String... queueNames) {
 		Assert.notNull(queueNames, "'queueNames' cannot be null");
 		Assert.noNullElements(queueNames, "'queueNames' cannot contain null elements");
-		return this.queueNames.removeAll(Arrays.asList(queueNames));
+		AtomicBoolean result = new AtomicBoolean();
+		if (this.queues.size() > 0) {
+			Set<String> toRemove = new HashSet<>(Arrays.asList(queueNames));
+			this.queues.stream().forEach(q -> {
+				if (toRemove.contains(q.getName()) || toRemove.contains(q.getDeclaredName())) {
+					this.queues.remove(q);
```

Comment:
Is it really safe to remove from the collection during its iteration?
Seems for me in other places we use `Iterator` on the matter.

On the other hand there is a `removeIf()` on the `Collection`:
```
 /**
     * Removes all of the elements of this collection that satisfy the given
     * predicate.  Errors or runtime exceptions thrown during iteration or by
     * the predicate are relayed to the caller.
```

Reply:
Good catch.